domains/transfer/Transfer.API/API/Transfer/Api/v2023_11_23/Controllers/TransferAgreementProposalController.cs

```diff
@@ -0,0 +1,145 @@
+        this.createTransferAgreementProposalValidator = createTransferAgreementProposalValidator;
+        this.transferAgreementRepository = transferAgreementRepository;
+    }
+
+    /// <summary>
+    /// Create TransferAgreementProposal
+    /// </summary>
+    /// <param name="request">The request object containing the StartDate, EndDate and ReceiverTin needed for creating the Transfer Agreement.</param>
+    /// <response code="201">Created</response>
+    /// <response code="400">Bad request</response>
+    /// <response code="409">There is already a Transfer Agreement with this company tin within the selected date range</response>
+    [ProducesResponseType(typeof(Guid), 201)]
+    [ProducesResponseType(typeof(void), 400)]
+    [ProducesResponseType(typeof(string), 409)]
+    [HttpPost]
+    public async Task<ActionResult> CreateTransferAgreementProposal(CreateTransferAgreementProposal request)
+    {
+        var companySenderId = Guid.Parse(User.FindSubjectGuidClaim());
+        var companySenderTin = User.FindSubjectTinClaim();
+        var companySenderName = User.FindSubjectNameClaim();
+
+        var validateResult = await createTransferAgreementProposalValidator.ValidateAsync(request);
+        if (!validateResult.IsValid)
+        {
+            validateResult.AddToModelState(ModelState);
+            return ValidationProblem(ModelState);
+        }
+
+        var newProposal = new TransferAgreementProposal
+        {
+            SenderCompanyId = companySenderId,
+            SenderCompanyTin = companySenderTin,
+            SenderCompanyName = companySenderName,
+            Id = Guid.NewGuid(),
+            ReceiverCompanyTin = request.ReceiverTin,
+            StartDate = DateTimeOffset.FromUnixTimeSeconds(request.StartDate),
+            EndDate = request.EndDate == null ? null : DateTimeOffset.FromUnixTimeSeconds(request.EndDate.Value)
+        };
+
+        if (request.ReceiverTin != null)
+        {
+            var hasConflict = await transferAgreementRepository.HasDateOverlap(newProposal);
+
+            if (hasConflict)
+            {
+                return ValidationProblem("There is already a Transfer Agreement with this company tin within the selected date range", statusCode: 409);
+            }
+        }
+
+        await repository.AddTransferAgreementProposal(newProposal);
+
+        return CreatedAtAction(nameof(GetTransferAgreementProposal), new { id = newProposal.Id }, newProposal);
+    }
+
+    /// <summary>
+    /// Get TransferAgreementProposal by Id
+    /// </summary>
+    /// <param name="id">Id of TransferAgreementProposal</param>
+    /// <response code="200">Successful operation</response>
+    /// <response code="400">You cannot Accept/Deny your own TransferAgreementProposal, you cannot Accept/Deny a TransferAgreementProposal for another company or this proposal has run out</response>
+    /// <response code="404">TransferAgreementProposal expired or deleted</response>
```

Comment:
Do we want to expose these details in 404?

Reply:
What do you suggest instead?

Reply:
corrected